### PR TITLE
add alias for non-existing ns

### PR DESCRIPTION
**Pull request by madstap**

### CODE REVIEW

File: src/com/gfredericks/schpec.clj [hunk 1]
```diff
@@ -44,3 +45,11 @@
                   [name `(s/and ~spec ~@preds)]))
               key-pred-forms)]
     `(s/or ~@opts)))
+
+(defmacro alias
+  "Like clojure.core/alias, but can alias to non-existing namespaces"
+  [alias namespace-sym]
+  `(try (clojure.core/alias ~alias ~namespace-sym)
+        (catch Exception _#
+          (do (create-ns ~namespace-sym)
```
Comment (gfredericks) [inline src/com/gfredericks/schpec.clj]: the `do` here is redundant -- `catch` can already take multiple expressions

File: test/com/gfredericks/test_schpec.clj [hunk 2]
```diff
@@ -41,3 +41,10 @@
   (is (nil? (s/explain-data ::my-xor {::y 1})))
   (is (some? (s/explain-data ::my-xor {::x 1 ::y 1})))
   (is (some? (s/explain-data ::my-xor {::y "abc"}))))
+
+(schpec/alias 'p 'person)
+
+(deftest alias-test
+  (let [aliases (ns-aliases 'com.gfredericks.test-schpec)]
```
Comment (gfredericks) [inline test/com/gfredericks/test_schpec.clj]: this makes me wonder if it's possible to register an alias at a lower level without having to create the namespace? That might be good for tools that enumerate through all the namespaces and might get confused

Comment (gfredericks) [reply test/com/gfredericks/test_schpec.clj]: it looks like there's no reasonably way of doing that

File: src/com/gfredericks/schpec.clj [hunk 3]
```diff
@@ -44,3 +45,11 @@
...
               key-pred-forms)]
     `(s/or ~@opts)))
+
+(defmacro alias
+  "Like clojure.core/alias, but can alias to non-existing namespaces"
+  [alias namespace-sym]
+  `(try (clojure.core/alias ~alias ~namespace-sym)
+        (catch Exception _#
+          (create-ns ~namespace-sym)
+          (clojure.core/alias ~alias ~namespace-sym))))
```
Comment (gfredericks) [inline src/com/gfredericks/schpec.clj]: Is there any need for this to be a macro? If not I'd rather keep it as a function to match the signature of `clojure.core/alias`.

**Review by gfredericks — CHANGES_REQUESTED** [review]
this should be good to merge after this change

File: src/com/gfredericks/schpec.clj (hunk 3 above)
Comment (madstap) [reply src/com/gfredericks/schpec.clj]: Yes, of course. For some reason I thought `clojure.core/alias` was a macro...

**Review by gfredericks — APPROVED** [review]

**Comment by gfredericks** [comment]
this is great, thanks!

**Comment by gfredericks** [comment]
Released in 0.1.1.

**Pull request merged**